packages/mc-scripts/src/commands/deployments-push.ts

```diff
@@ -0,0 +1,253 @@
+import chalk from 'chalk';
+import prompts from 'prompts';
+import {
+  processConfig,
+  type CustomApplicationData,
+  type CustomViewData,
+} from '@commercetools-frontend/application-config';
+import type { TCliCommandPushDeploymentPreviewOptions } from '../types';
+import CredentialsStorage from '../utils/credentials-storage';
+import {
+  fetchCustomApplication,
+  updateCustomApplicationDeploymentPreview,
+  createCustomApplicationDeploymentPreview,
+} from '../utils/graphql-requests';
+
+const credentialsStorage = new CredentialsStorage();
+
+const deploymentPreviewAliasRegex = /^[^-#]([a-z]|[-](?![-])){0,62}[^-#]$/g;
+export const fqdnRegex =
+  /(?=^.{4,253}$)(^((?!-)[a-zA-Z0-9-]{0,62}[a-zA-Z0-9]\.)+[a-zA-Z]{2,63}$)/i;
```

Comment:
These ones were copied from the MC frontend.
I would suggest to move them to the constants package and use them from there both here and in the frontend.

On the other hand, I think we should adjust the RegExps or even think about what we want to validate as current ones deny some values I find legit:
```
// We're currently blocking aliases like these:
pr-1234
branch-1234
team-1234
v2-branch

// We're currently blocking URLs like these:
https://pr_1234.surge.sh

Reply:
Simplified validations according to the agreement with @emmenko: e624165c7d4d0aaa07e2a0e0c8004d380171d750